### PR TITLE
feat: Remove deprecated node-sass package, migrated to Sass #2392

### LINKED ISSUE #2392: Deprecated package `node-sass`
We are compiling `.scss` files to `.css` using the `node-sass` package which is [deprecated](https://www.npmjs.com/package/node-sass).

We should switch over to `sass` (Dart Sass).

* https://sass-lang.com/dart-sass/
* https://www.npmjs.com/package/sass

This will involve installing the correct packages but will also likely involve changes to our Webpack config files.

_Originally posted by @lindapaiste in https://github.com/processing/p5.js-web-editor/pull/2388#pullrequestreview-1585040708_

### PR DESCRIPTION
**Pull request by luckyklyist**
This PR addresses issue #2392.

**Changes Made:** #2392  

- Removed the deprecated "node-sass" package.
- Added the "sass" package as a replacement.

**Details:**

In this PR, I've removed the deprecated "node-sass" package and replaced it with the "sass" package, which is the recommended alternative. This update ensures that we are using the latest and supported Sass compiler.

To migrate the deprecated "node-sass" syntax code to the new "sass" syntax, I used the [Sass Migrator](https://sass-lang.com/documentation/cli/migrator/), which automates the migration process. 

* [x] has no linting errors (`npm run lint`)
* [ ] has no test errors (`npm run test`)
* [x] is from a uniquely-named feature branch and is up to date with the  `develop` branch.
* [x] is descriptively named and links to an issue number, i.e. `Fixes #2392 `

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,7 +23,8 @@
     "update-syntax-highlighting": "node ./server/scripts/update-syntax-highlighting.js",
     "update-p5-hinter": "node ./server/scripts/update-p5-hinter.js",
     "heroku-postbuild": "touch .env; npm run build",
-    "storybook:build:css": "node-sass client/styles/main.scss client/styles/storybook.css",
+    "storybook:build:css": "sass client/styles/main.scss client/styles/storybook.css",
+    "migrate-sass": "sass-migrator division 'client/styles/**/*.scss'",
```
Comment (lindapaiste) [inline package.json]: Thanks so much for working on this!  The migration is a one-time process, right?  So we can probably remove this command now that it is complete?

**Review by lindapaiste — CHANGES_REQUESTED** [review]

**Comment by lindapaiste** [comment]
🥇 Just wanted to say that this is a very good PR!   You've explained what you changed and why/how.  Much appreciated.

**Comment by luckyklyist** [comment]
> 🥇 Just wanted to say that this is a very good PR! You've explained what you changed and why/how. Much appreciated.

Thanks for the praise! It's greatly appreciated!  Looking forward to your review and the PR getting merged!

File: package.json (hunk 1 above)
Comment (luckyklyist) [reply package.json]: Yes .

**Comment by luckyklyist** [comment]
@lindapaiste I have removed the sass-migrator from the package.json scripts  .

**Comment by luckyklyist** [comment]
hey @lindapaiste  is there any other things to change in this pr ?

**Comment by lindapaiste** [comment]
This is looking great! I just ran the branch and had no problems running in dev.

There is one tiny thing to address which is the docker scripts, where we have an `npm rebuild node-sass` command.
https://github.com/processing/p5.js-web-editor/blob/ca74574cd1830d900aad12665f15fc5f09251f14/Dockerfile#L13
https://github.com/processing/p5.js-web-editor/blob/ca74574cd1830d900aad12665f15fc5f09251f14/Dockerfile#L30

I'm not sure what's the correct thing to put here when using Dart Sass. I think nothing?  Like we delete those lines with no replacement?  Do you know?

**Comment by luckyklyist** [comment]
I don't think there is rebuild command for the Dart Sass as it doesn't rely on native add-ons .I have remove the node-sass rebuild command from the dockerFile .

**Review by lindapaiste — DISMISSED** [review]

**Comment by lindapaiste** [comment]
@raclim this looks good to me but I’m a little bit scared to merge it as it has the potential to break the built process.

**Review by lindapaiste — APPROVED** [review]
@raclim I fixed all the conflicts so I'm going to go ahead and merge this in.

**Comment by raclim** [comment]
@lindapaiste Thanks so much for getting this in!

**Comment by lindapaiste** [comment]
@raclim just a heads up that we may have some pending PRs using the old syntax for the rem font size.  When merging those, we'll want to change `#{20 / $base-font-size}rem;` etc to `#{math.div(20, $base-font-size)}rem;`.  I think the older syntax will still work but it's now deprecated.

**Pull request merged**